### PR TITLE
compile and run on Windows

### PR DESCRIPTION
**Pull request by mrexodia**
Currently it is linked to a submodule https://github.com/mrexodia/iaito_win32 which I can transfer to @hteso once everything is considered stable.

- **You need https://github.com/radare/radare2/commit/cd28744049492dc8ac25a1f2b3ba0e42f0e9ce93 to actually run this.**

---
This change is [Reviewable](https://reviewable.io/reviews/hteso/iaito/24)

### CODE REVIEW

**Comment by xarkes** [comment]
It seems quite heavy to me to clone another git repo just to compile it under Windows. The headers already are in the radare2 repository and I think it should not be very hard to include compiled libraries the same way as we do on linux.

**Comment by mrexodia** [comment]
I did that to not dirty the repo with lib files and msvc-dependent hacks.
Obviously I could directly add the lib files and link to radare as a
submodule but this is by far the prettiest way.

**Comment by xarkes** [comment]
I think it's not a problem to have msvc-dependent hacks in the iaito repository. Also I don't understand the problem with the lib files, there's no such problem with linux .so files

**Comment by mrexodia** [comment]
It's just that they have to be manually regenerated if r2 is updated but
sure I can move the files from the submodule to the repo itself...

**Comment by xarkes** [comment]
Yes but that is how it works currently for Mac Os and Linux, why should it be different ? You compile first radare2 then you compile iaito. If you want updated libraries then you compile radare2 again.
Anyway this is just my own opinion about having a (useless to my mind) sub git repository. If you have good reasons to keep it, then keep it.

**Comment by mrexodia** [comment]
No you don't. Radare doesn't provide libraries. You have to run some
scripts to hack around that and generate the lib files that visual studio
needs. On Windows you also don't have /usr/lib/radare2 or whatever

**Comment by trufae** [comment]
I hate pushing binary files into git. Its a very bad practice that only windows users do. If you insist on that i would suggest to use git lfs or a submodule repository which will be automatilly fetched

**Comment by trufae** [comment]
Windows build should have relative paths for all that. If i missed any pls fill an issue in the radare repo.

Also, there are env vars and config vars to change the paths of plugins, databases, etc

**Comment by mrexodia** [comment]
> or a submodule repository which will be automatilly fetched

That's exactly what I did 👍 

> Windows build should have relative paths for all that.

Yeah but you have to point the Qt `.pro` to the r2 files. I currently did this by adding an include path with a relative path which points to the submodule. This allows Windows users to compile out of the box (r2 uses `<sys/time.h>` which doesn't exist so that's why I did it like this).

**Comment by mrexodia** [comment]
Okay @hteso I made some changes and everything compiles on MinGW too. It can be merged I think.

I have also added an entry to [the wiki](https://github.com/hteso/iaito/wiki/Compiling-on-Windows).

**Comment by ioncodes** [comment]
In my opinion @mrexodia his solution is the best compared to the other ideas in the comments.

**Comment by trufae** [comment]
Fixing this sys/time thing is easy. But i have no way to do that without windows. Also maintaining more than one build system is a pain. So it will be good if we can have r2 compiled natively with visual studio preferibly in docker

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -244,7 +245,7 @@ void MainWindow::start_web_server() {
     // Start web server
     thread.core = core;
     thread.start();
-    sleep (1);
```
Comment (radare) [inline src/mainwindow.cpp:247]: You can also use r_sys_sleep, which us portable way using r2 api but i think its better to stick with the qt apis

File: src/qrcore.cpp [hunk 2]
```diff
@@ -130,34 +130,32 @@ QRCore::~QRCore() {
...
     if (va==0 || va == 2)
         r_config_set_i (core->config, "io.va", va);
     // NO ONE KNOWS WHY THIS IS FIXING A SEGFAULT. core->file should have already a proper value. Pancake dixit
     //core->file = NULL;
     // mapaddr = 0LL;
-    char *paz = (char*)path.toStdString().c_str();
-    //printf ("FILE OPEN (%s)\n", paz);
-    f = r_core_file_open(core, path.toUtf8(), rw?(R_IO_READ|R_IO_WRITE):R_IO_READ, mapaddr);
+    printf ("FILE OPEN (%s)\n", path.toUtf8().constData());
+    f = r_core_file_open(core, path.toUtf8().constData(), rw?(R_IO_READ|R_IO_WRITE):R_IO_READ, mapaddr);
```
Comment (radare) [inline src/qrcore.cpp:150]: Missing space before (

Comment (radare) [reply src/qrcore.cpp:150]: Which coding style are we gonna use in iaito? Default from qtcreator? The same in r2? Define an uncrustify profile?

**Comment by radare** [comment]
This PR lgtm and fixes a lot of things. So its ok for me to merge

**Comment by mrexodia** [comment]
I don't really care about the spaces but it was inconsistent at best so I
just typed what I usually do. Personally I think we should stick to Qt's
style but I really don't care.

**Comment by trufae** [comment]
Its ok to me if we just using the same rules for all the project

**Pull request merged**